### ISSUE TITLE
Bug: Request packet sent is bigger than negotiated PDU size

### ISSUE THREAD

**Issue by gfcittolin** [plcpeople/nodeS7#40]
*TL;DR*
We're not caring about the request packet size. PR suggested
---

This is the result of the analysis of the following issue: netsmarttech/node-red-contrib-s7#10

The issue there says that, although all variables seem to be setup correctly, the read process return "Bad values". With the variable list provided and the logs, I could reproduce the issue locally (with Snap7 acting as the PLC).

With the help of Wireshark, we can take a look at the packets exchanged and I found that, even though the negotiated PDU size was 240 bytes... 
![image](https://user-images.githubusercontent.com/2856088/29498036-79e74ca0-85ca-11e7-843e-003e020fb087.png)
... `nodes7` was requesting a packet bigger than that (in this case, 276 bytes - can be seen in the status bar):
![image](https://user-images.githubusercontent.com/2856088/29498050-b1fb3dfe-85ca-11e7-9d03-d6f27750379d.png)

Checking on how the packets are created, I saw that care is taken so that the **reply** is not bigger than the PDU size, but not the **request**. This usually not the case, but can occur under the following circumstance:
 - The variables span across multiple areas or multiple DBs, so we cannot optimize under a single memory are read operation, and
 - The length being read is very small, so that a whole read command (which has a fixed length of 12 bytes) is bigger than its response (which size is normally `data_length + 4`), and
- There are lots of these variables, so we overflow the request packet

I propose to fix this by computing the resulting request packet size, not only the reply one. Then we can split it when the response _or when the request_ would be bigger than the negotiated PDU size. I've implemented this locally and it seems to solve the problem. A PR will follow with this suggestion.

**Comment by plcpeople** [comment]
Good catch.  I obviously had not considered this possiblity in the original code.
